synapse/assets/storm/migrations/model-0.2.27.storm

```diff
@@ -0,0 +1,348 @@
+
+            spin |
+
+            iden $iden |
+
+            // Get sources and store them in the queue
+            // Do this before we store/delete edges.
+            { <(seen)- meta:source
+                $sources.append($node.repr())
+            }
+
+            // Get N1 edges and store them in the queue
+            { for $edge in $node.edges() {
+                ($verb, $dst) = $edge
+                $edges.append($edge)
+
+                [ -($verb)> { iden $dst } ]
+
+                // Flush to the queue if the list grows too large
+                if ($edges.size() > 1000) { // pragma: no cover start
+                    $item = ({
+                        "iden": $iden,
+                        "view": $view.iden,
+                        "direction": "n1",
+                        "edges": $edges,
+                    })
+
+                    $edgeq.put($item)
+                    $edges = ([])
+                } // pragma: no cover stop
+            }}
+
+            if $edges {
+                $item = ({
+                    "iden": $iden,
+                    "view": $view.iden,
+                    "direction": "n1",
+                    "edges": $edges,
+                })
+
+                $edgeq.put($item)
+                $edges = ([])
+            }
+
+            // Get N2 edges and store them in the queue
+            { for $edge in $node.edges(reverse=$lib.true) {
+                ($verb, $src) = $edge
+                $edges.append($edge)
+
+                [ <($verb)- { iden $src } ]
+
+                // Flush to the queue if the list grows too large
+                if ($edges.size() > 1000) { // pragma: no cover start
+                    $item = ({
+                        "iden": $iden,
+                        "view": $view.iden,
+                        "direction": "n2",
+                        "edges": $edges,
+                    })
+
+                    $edgeq.put($item)
+                    $edges = ([])
+                } // pragma: no cover stop
+            }}
+
+            if $edges {
+                $item = ({
+                    "iden": $iden,
+                    "view": $view.iden,
+                    "direction": "n2",
+                    "edges": $edges,
+                })
+
+                $edgeq.put($item)
+                $edges = ([])
+            }
+        }
+    }
+
+    $item = ({
+        "iden": $iden,
+        "form": $form,
+        "repr": $repr,
+        "props": $n.props,
+        "view": $viewiden,
+        "layer": $layriden,
+        "tags": $n.tags(),
+        "data": $n.data.list(),
+        "sources": $sources,
+    })
+
+    $nodeq.put($item)
+
+    return()
+}
+
+function migrateCpeNode(oldcpe, newcpe) {
+
+    // migrate the edges, tags, and data
+    $lib.model.migration.copyEdges($oldcpe, $newcpe)
+    $lib.model.migration.copyTags($oldcpe, $newcpe)
+    $lib.model.migration.copyData($oldcpe, $newcpe)
+
+    $refinfos = $getRefInfo($oldcpe.form())
```

Comment:
Could we do this once as a module local?

Reply:
No, because it's per form we're trying to delete.

Reply:
I could probably cache it per form though.

Reply:
Oh, you meant just the `it:sec:cpe` one. Yes, we probably could. I added a cache mechanism to that function so it should only happen once now.